odoo/addons/base/models/res_company.py

```diff
@@ -340,6 +340,15 @@ def unlink(self):
         return res
 
     def write(self, values):
+        if 'parent_id' in values:
+            raise UserError(_("The company hierarchy cannot be changed."))
+
+        if values.get('currency_id'):
+            currency = self.env['res.currency'].browse(values['currency_id'])
+            if not currency.active:
```

Comment:
You may have a MissingError here. But not important since no one complains.

Reply:
MissingError is usually a programming error. I don't think we need to do anything for it.